### ISSUE TITLE
pydotfiles v0.0.4 compatibility

### ISSUE THREAD

**Issue by JasonYao** [JasonYao/dotfiles#6]
## (7) Issue 7
- issue: had to type in password twice
```
[USER] Sudo: Actions requiring sudo actions have been detected, please enter your password below (your password is NOT stored at any point in time)
Password: 
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Install: Installing module [name=symlinks]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/gitconfig.symlink -> /Users/jason/.gitconfig]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/gitconfig.symlink -> /Users/jason/.gitconfig]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/servers.symlink -> /Users/jason/.servers]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/servers.symlink -> /Users/jason/.servers]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/colour.symlink -> /Users/jason/.colour]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/colour.symlink -> /Users/jason/.colour]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/specific.symlink -> /Users/jason/.specific]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/specific.symlink -> /Users/jason/.specific]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/iterm2_shell_integration.bash.symlink -> /Users/jason/.iterm2_shell_integration.bash]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/iterm2_shell_integration.bash.symlink -> /Users/jason/.iterm2_shell_integration.bash]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/vimrc.symlink -> /Users/jason/.vimrc]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/vimrc.symlink -> /Users/jason/.vimrc]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/motd.symlink -> /Users/jason/.motd]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/motd.symlink -> /Users/jason/.motd]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/inputrc.symlink -> /Users/jason/.inputrc]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/inputrc.symlink -> /Users/jason/.inputrc]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/bashrc.symlink -> /Users/jason/.bashrc]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/bashrc.symlink -> /Users/jason/.bashrc]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/env.symlink -> /Users/jason/.env]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/env.symlink -> /Users/jason/.env]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/utilities.symlink -> /Users/jason/.utilities]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Action: Successfully completed action [Action=SYMLINK /Users/jason/.dotfiles/symlinks/utilities.symlink -> /Users/jason/.utilities]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Install: Successfully installed module [name=symlinks]
2018-11-05 10:38:41 PM [INFO::pydotfiles.models]: Install: Installing module [name=macos]
  [ OK ] Keyboard: Press & hold has already been disabled
  [ OK ] Keyboard: 0-delay key repeat has already been enabled
  [ OK ] Keyboard: 0-delay initial key repeat has already been enabled
  [ OK ] Finder: Show hard drives on desktop is already enabled
  [ OK ] Finder: Show external hard drives on desktop is already enabled
  [ OK ] Finder: Show removable media on desktop is already been enabled
  [ OK ] Battery: Show percent battery remaining is already enabled
  [ .. ] Battery: Disabling computer/display sleep
Password:
setcomputersleep: Never
setdisplaysleep: Never
```

**Comment by JasonYao** [comment]
(10) minor issue

fuck autocorrect, turn this setting off

**Comment by JasonYao** [comment]
### (9) Gradle issue
- issue: installing gradlew is borked, looks like I need to install java first before I can install gradle
```
2018-11-05 10:51:56 PM [INFO::pydotfiles.models]: Package: Attempting to install `gradle`
Traceback (most recent call last):
  File "/usr/local/var/pyenv/versions/default-3.7.0/bin/pydotfiles", line 13, in <module>
    main(sys.argv)
  File "/usr/local/var/pyenv/versions/default-3.7.0/bin/pydotfiles", line 9, in main
    dispatcher.dispatch()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/api/__init__.py", line 48, in dispatch
    getattr(self, args.command)(command_arguments)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/api/__init__.py", line 98, in install
    self.dotfiles.install_all()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 112, in install_all
    module.install()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 213, in install
    self.operating_system.install_packages()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 374, in install_packages
    self.install_package(package)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 424, in install_package
    raise NotImplementedError(f"Package: Failed to install package `{package}` [stderr={stderr}]")
NotImplementedError: Package: Failed to install package `gradle` [stderr=gradle: Java 1.7+ is required to install this formula.
JavaRequirement unsatisfied!
You can install with Homebrew Cask:
 brew cask install java
You can download from:
 https://www.oracle.com/technetwork/java/javase/downloads/index.html
Error: An unsatisfied requirement failed this build.
]
          [FAIL] Pydotfiles: Failed to install the pydotfiles configurations
```

**Comment by JasonYao** [comment]
### (15) ssh config
- issue: the `~/.ssh` directory doesn't exist
```
2018-11-05 11:48:20 PM [INFO::pydotfiles.models]: Action: Starting action [Action=SYMLINK /Users/jason/.dotfiles/ssh/config.symlink -> /Users/jason/.ssh/config]
Traceback (most recent call last):
  File "/usr/local/var/pyenv/versions/default-3.7.0/bin/pydotfiles", line 13, in <module>
    main(sys.argv)
  File "/usr/local/var/pyenv/versions/default-3.7.0/bin/pydotfiles", line 9, in main
    dispatcher.dispatch()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/api/__init__.py", line 48, in dispatch
    getattr(self, args.command)(command_arguments)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/api/__init__.py", line 98, in install
    self.dotfiles.install_all()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 112, in install_all
    module.install()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 219, in install
    self.do_actions()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 258, in do_actions
    override_action = self.__do_action_with_override__(override_action, action)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/__init__.py", line 279, in __do_action_with_override__
    action.do()
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/models/primitives.py", line 62, in do
    symlink_file(self.origin, self.destination, self.run_as_sudo, self.sudo_password)
  File "/usr/local/var/pyenv/versions/3.7.0/envs/default-3.7.0/lib/python3.7/site-packages/pydotfiles/utils/io.py", line 102, in symlink_file
    os.symlink(origin, destination)
FileNotFoundError: [Errno 2] No such file or directory: '/Users/jason/.dotfiles/ssh/config.symlink' -> '/Users/jason/.ssh/config'
          [FAIL] Pydotfiles: Failed to install the pydotfiles configurations
```
 issue is probably in the way the script is setup

**Comment by JasonYao** [comment]
### (16) default config
- issue: default doesn't exist
- other issue: need to add in support for dark mode on mojave+
```
 [ OK ] Spotlight: External disk indexing is already disabled
2018-11-05 23:55:05.476 defaults[20619:299769] 
The domain/default pair of (/Volumes/MacOS/.Spotlight-V100/VolumeConfiguration.plist, Exclusions) does not exist
2018-11-05 23:55:05.524 defaults[20621:299776] Could not write domain /Volumes/MacOS/.Spotlight-V100/VolumeConfiguration.plist; exiting
  [ OK ] Spotlight: Bootcamp partition has been excluded from normal indexing
  [ OK ] Mission Control: Automatic rearranging spaces has already been disabled
  [ OK ] DNS: Host's DNS servers already point at Google

```

```
  [ OK ] Dock: All non-default applications are now sanitised
  [ .. ] Dock: Checking for all default app existences
2018-11-05 23:55:04.862 defaults[20563:299558] 
The domain/default pair of (com.apple.Dock, persistent-apps) does not exist
  [ .. ] Dock: Launchpad is not set on the dock, setting now
  [ OK ] Dock: Launchpad is now set on the dock
  [ .. ] Dock: Notes is not set on the dock, setting now

```

```
  [ OK ] Finder: Show removable media on desktop is already been enabled
  [ OK ] Battery: Show percent battery remaining is already enabled
  [ .. ] Battery: Disabling computer/display sleep
setcomputersleep: Never
setdisplaysleep: Never
  [ OK ] Battery: Successfully disabled computer/display sleep
  [ OK ] Trackpad: Tracking speed is already max

```

**Comment by JasonYao** [comment]
## Punchlist
- Add in default config turning off auto correct for notes
- Clean up default settings for problem 16

**Comment by JasonYao** [comment]
Using https://ss64.com/ is a good place to start for those commands, especially if I'm going to start tagging this repo (e.g. 10.14-mojave)

**Comment by JasonYao** [comment]
Fixed with #22
